src/custom/hooks/usePriceImpact/index.ts

```diff
@@ -151,26 +177,42 @@ export function useAbaPriceImpact({ abTrade, fiatPriceImpact }: AbaPriceImpactPa
     toDecimals: isExactIn ? abTrade?.inputAmount.currency.decimals : abTrade?.outputAmount.currency.decimals,
   })
 
-  // ref to not recalc useEffect
-  const abaPriceImpactRef = useRef<Percent | undefined>()
+  const [priceImpact, setPriceImpact] = useState<Percent | undefined>()
+
+  // we set price impact to undefined when loading
+  const quoteLoading = useIsQuoteLoading()
+
+  // primitive values to use as dependencies
+  const abIn = abTrade?.inputAmount.quotient.toString()
+  const abOut = abTrade?.outputAmount.quotient.toString()
+  const baOut = baTrade?.outputAmount.quotient.toString()
 
   useEffect(() => {
     // we have no fiat price impact and there's a trade, we need to use ABA impact
-    if (abTrade && baTrade && !fiatPriceImpact) {
-      const abaPriceImpact = _calculateAbaPriceImpact(abTrade, baTrade, isExactIn)
-      abaPriceImpactRef.current = abaPriceImpact
+    if (abIn && abOut && baOut) {
+      const abaPriceImpact = _calculateAbaPriceImpact(isExactIn ? abIn : abOut, baOut)
+      setPriceImpact(abaPriceImpact)
     }
-  }, [fiatPriceImpact, abTrade, baTrade, isExactIn])
+  }, [abIn, abOut, baOut, isExactIn])
 
-  return abaPriceImpactRef.current
+  // quote loading so we null the aba impact
+  // prevents lingering calculations and jumping impacts
+  useEffect(() => {
+    if (quoteLoading) {
+      setPriceImpact(undefined)
+    }
+  }, [quoteLoading])
+
+  return priceImpact
 }
 
 type PriceImpactParams = Omit<AbaPriceImpactParams, 'fiatPriceImpact'> & { parsedAmounts: ParsedAmounts }
 
 export default function usePriceImpact({ abTrade, parsedAmounts }: PriceImpactParams) {
```

Comment:
I think this function, and the re-export of all the other hooks, could be the only function in the index.ts